File: PA5WF/ExceptionHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PA5WF
{
    class ExceptionHandler
    {

		// Method that handles our exceptions, it returns an errormessage in the form of a string depending on what the errormessage contains

        public string HandleException(Exception ex)
        {
            //Error messege for primary key violation

            if (ex.Message.Contains("Violation of PRIMARY KEY"))
            {
                return "A person with these initials already exist please try another one.";
            }

            // Error message if there is no connection

            else if (ex.Message.Contains("error: 40 "))
			{
				return "Check your internet connection or contact our IT support at: 0730880819";
			}

            // Error message for failing to access database

            else if (ex.Message.Contains("Cannot open database"))
			{
				return "Connection failed to the database, please check if the name is spelled correctly for the database" + "\n or if the username and/or password is correct, or if the user has access to the database";
			}

            // Error message for server connection

            else if (ex.Message.Contains("error: 26"))
			{
				return "Connection failed to the server please check if you are connected to the server.";
			}

            //Error messege for exceeding character limit

            else if (ex.Message.Contains("System.Data.Entity.Validation.DbEntityValidationException"))
			{
				return "The amount of characters exceeds the limit of the database, the maximum amount of characters for First Name, Last Name, Job Title is 30 characters, the maximum amount for Address is 50.";
			}

            //Error messege when unknown error

            else
            {
                return "Try to restart the program if it is still not working please call our customer service at: 0730880819";
            }
			
        }
    }
}

File: PA5Java/src/pa5/ExceptionHandler.java
package pa5;

public class ExceptionHandler {
	
	// Method for handling exceptions 	
	public String HandleException(Exception ex) {
		
		if(ex instanceof NullPointerException){
			return "There were problems connecting to the server" + "\n please check your internet connection and/or " + "\n contacting IT support at: 0730880819"; 
		}
		// Error message for primary key violation
		else if(ex.getMessage().contains("Violation of PRIMARY KEY"))
        {
            return "A person with these initials" + "\nalready exist please try another one.";
        }
		// Error message for connection 
		else if (ex.getMessage().contains("error: 40 "))
		{
			return "Check your internet connection" +  "\nor contact our IT support at: 0730880819";
		}
		// Error message for failing to access database 
		else if (ex.getMessage().contains("Cannot open database"))
		{
			return "Connection failed to the database," + "\nplease check if the name is spelled correctly" +"\nfor the database or if the username" +"\nand/or password is correct," + "\nor if the user has access to the database";
		}
		// Error message for server connection
		else if (ex.getMessage().contains("error: 26"))
		{
			return "Connection failed to the server please" + "\n check if you are connected to the server.";
		}
		else if (ex.getMessage().contains("System.Data.Entity.Validation.DbEntityValidationException")) {
			return "The amount of characters exceeds the database limit, \n the maximum amount of characters is 30 for first name, \n last name and job title, and 50 for address";
		}
		// Error message for unexpected error
		else
        {
            return "Unknown error contact our IT support at: 0730880819";
        }
	}

}
